src/ReachSets/discretize.jl

```diff
@@ -887,3 +887,14 @@ function _discretize_interval_matrix_inhomog(U, Ω0_homog, linear_maps,
     end
     return Ω0, Ud
 end
+
+# fallback implementation for conversion (if applicable) or overapproximation
+function _convert_or_overapproximate(X::Type{<:Zonotope}, Y::LazySet)
+    if applicable(convert, X, Y)
+        return convert(X, Y)
+    elseif applicable(overapproximate, Y, X)
+        return overapproximate(Y, X)
+    else
+        return convert(Zonotope, overapproximate(Y, Hyperrectangle))
```

Comment:
```suggestion
        return convert(X, overapproximate(Y, Hyperrectangle))
```
(change needed for the generalization)

It is nice to make discretization work with arbitrary sets, but an option to choose this approximation would be nice.

Reply:
> It is nice to make discretization work with arbitrary sets, but an option to choose this approximation would be nice.

Thinking about this, this function is actually general enough to be a feature for `LazySets.Approximations`.

Reply:
It's a rather trivial function, i think it's good here..

Reply:
In general if you use `overapproximate` we'd like that it converts if it can (there is an open issue for that). This PR was specific for *zonotopes*.

Reply:
> It's a rather trivial function

I do not see how that makes it more appropriate here.

> In general if you use overapproximate we'd like that it converts if it can (there is an open issue for that).

Good, because this function seems like the way to solve it.